nemo/collections/nlp/data/intent_slot_classification/intent_slot_classification_descriptor.py

```diff
@@ -0,0 +1,143 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+
+import itertools
+
+from nemo import logging
+from nemo.collections.nlp.data.data_utils.data_preprocessing import (
+    fill_class_weights,
+    get_freq_weights,
+    get_label_stats,
+    if_exist,
+)
+
+
+class IntentSlotDataDesc:
+    """ Convert the raw data to the standard format supported by
+    IntentSlotDataDesc.
+
+    By default, the None label for slots is 'O'.
+
+    IntentSlotDataDesc requires two files:
+
+        input_file: file to sequence + label.
+            the first line is header (sentence [tab] label)
+            each line should be [sentence][tab][label]
+
+        slot_file: file to slot labels, each line corresponding to
+            slot labels for a sentence in input_file. No header.
+
+    To keep the mapping from label index to label consistent during
+    training and inferencing we require the following files:
+        dicts.intents.csv: each line is an intent. The first line
+            corresponding to the 0 intent label, the second line
+            corresponding to the 1 intent label, and so on.
+
+        dicts.slots.csv: each line is a slot. The first line
+            corresponding to the 0 slot label, the second line
+            corresponding to the 1 slot label, and so on.
+
+    Args:
+        data_dir (str): the directory of the dataset
+        none_slot_label (str): the label for slots that aren't identified
+            defaulted to 'O'
+        pad_label (int): the int used for padding. If set to -1,
+             it'll be set to the whatever the None label is.
+    """
+
+    def __init__(self, data_dir, none_slot_label='O', pad_label=-1):
+        if not if_exist(data_dir, ['dict.intents.csv', 'dict.slots.csv']):
+            raise FileNotFoundError(
+                "Make sure that your data follows the standard format "
+                "supported by JointIntentSlotDataset. Your data must "
+                "contain dict.intents.csv and dict.slots.csv."
+            )
+
+        self.data_dir = data_dir
+        self.intent_dict_file = self.data_dir + '/dict.intents.csv'
+        self.slot_dict_file = self.data_dir + '/dict.slots.csv'
+
+        self.intents_label_ids = IntentSlotDataDesc.label2idx(self.intent_dict_file)
+        self.num_intents = len(self.intents_label_ids)
+        self.slots_label_ids = IntentSlotDataDesc.label2idx(self.slot_dict_file)
+        self.num_slots = len(self.slots_label_ids)
+
+        infold = self.data_dir
+        for mode in ['train', 'test', 'dev']:
```

Comment:
I have dropped the mode inputs and replaced them with full path of the files. It does not makes sense to have such limitation anymore. In the new PR, I updated text classification model. You may do it in the next round if you want: https://github.com/NVIDIA/NeMo/pull/910